tests/cancun/eip1153_tstore/test_tstore_reentrancy.py

```diff
@@ -0,0 +1,212 @@
+    tload_value_set_in_call = 90
+
+    # Storage cells
+    slot_tload_before_call = 0
+    slot_tload_in_subcall_result = 1
+    slot_tload_after_call = 2
+    slot_subcall_worked = 3
+    slot_tload_1_after_call = 4
+    slot_tstore_overwrite = 5
+    slot_code_worked = 6
+
+    # Function names
+    do_tstore = 1
+    do_reenter = 2
+
+    def make_call(call_type: Op) -> Bytecode:
+        if call_type == Op.DELEGATECALL or call_type == Op.STATICCALL:
+            return call_type(Op.GAS(), call_dest, 0, 32, 32, 32)
+        else:
+            return call_type(Op.GAS(), call_dest, 0, 0, 32, 32, 32)
+
+    subcall_code = (
+        Op.TSTORE(0, 89)
+        + Op.TSTORE(0, tload_value_set_in_call)
+        + Op.TSTORE(1, 11)
+        + Op.TSTORE(1, 12)
+        + Op.MSTORE(0, Op.TLOAD(0))
+        + call_return(0, 32)
+    )
+
+    pre = {
+        address_to: Account(
+            balance=1000000000000000000,
+            nonce=0,
+            code=Switch(
+                cases=[
+                    Case(
+                        condition=Op.EQ(Op.CALLDATALOAD(0), do_tstore),
+                        action=subcall_code,
+                    ),
+                    Case(
+                        condition=Op.EQ(Op.CALLDATALOAD(0), do_reenter),
+                        action=Op.TSTORE(0, tload_value_set_before_call)
+                        + Op.SSTORE(slot_tload_before_call, Op.TLOAD(0))
+                        + Op.MSTORE(0, do_tstore)
+                        + Op.MSTORE(32, 0xFF)
+                        + Op.SSTORE(slot_subcall_worked, make_call(call_type))
+                        + Op.SSTORE(slot_tload_in_subcall_result, Op.MLOAD(32))
+                        + Op.SSTORE(slot_tload_after_call, Op.TLOAD(0))
+                        + Op.SSTORE(slot_tload_1_after_call, Op.TLOAD(1))
+                        + Op.TSTORE(0, 50)
+                        + Op.SSTORE(slot_tstore_overwrite, Op.TLOAD(0))
+                        + Op.SSTORE(slot_code_worked, 1),
+                    ),
+                ],
+                default_action=None,
+            ),
+            storage={
+                slot_tload_before_call: 0xFF,
+                slot_tload_in_subcall_result: 0xFF,
+                slot_tload_after_call: 0xFF,
+                slot_subcall_worked: 0xFF,
+                slot_tload_1_after_call: 0xFF,
+                slot_tstore_overwrite: 0xFF,
+                slot_code_worked: 0xFF,
+            },
+        ),
+        address_code: Account(
+            balance=0,
+            nonce=0,
+            code=subcall_code,
+            storage={},
+        ),
+        TestAddress: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code="0x",
+            storage={},
+        ),
+    }
+
+    post: Dict[Address, Union[Account, object]] = {}
+
+    def failing_calls() -> bool:
+        return call_type == Op.STATICCALL or call_return == Op.REVERT or call_return == Om.OOG
```

Comment:
This can be a variable instead of a function.
```suggestion
    failing_calls = (call_type == Op.STATICCALL or call_return in [Op.REVERT, Om.OOG])
```